dist/go/stage.go

```diff
@@ -71,3 +78,28 @@ func (v *Stage) UnmarshalJSON(data []byte) error {
 
 	return json.Unmarshal(obj.Spec, v.Spec)
 }
+
+func (v1 *StageV1) UnmarshalJSONV1(data []byte) error {
+	type TempStage struct {
+		Name    string          `json:"name,omitempty"`
+		Clone   *CloneStageV1   `json:"clone,omitempty"`
+		Runtime string     		`json:"runtime,omitempty"`
+		Steps   []*StepV1       `json:"steps,omitempty"`
+	}
+
+	temp := &TempStage{}
+	if err := json.Unmarshal(data, temp); err != nil {
+		return err
+	}
+
+	v1.Name = temp.Name
+	if temp.Clone == nil {
+		v1.Clone = &CloneStageV1{Disabled: true}
+	} else {
+		v1.Clone = temp.Clone
+	}
+	v1.Runtime = "machine"
```

Comment:
we should choose if it is hosted or on runner.

Reply:
We have runtime defined and now it will choose whatever the value will be and the default value will be `machine`.